flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlChunkSplitter.java

```diff
@@ -179,7 +179,7 @@ private MySqlSnapshotSplit splitOneUnevenlySizedChunk(TableId tableId) throws SQ
                         chunkSize);
         // may sleep a while to avoid DDOS on MySQL server
         maySleep(nextChunkId, tableId);
-        if (chunkEnd != null && ObjectUtils.compare(chunkEnd, minMaxOfSplitColumn[1]) <= 0) {
+        if (chunkEnd != null) {
```

Comment:
The `chunkEnd` is possible to be larger than the recorded max value. Deleting this condition will cause endless processing.